drizzle-kit/src/cli/connections.ts

```diff
@@ -655,3 +656,82 @@ export const connectToSQLite = async (
 	);
 	process.exit(1);
 };
+
+const parseSingleStoreCredentials = (credentials: SingleStoreCredentials) => {
+	if ('url' in credentials) {
+		const url = credentials.url;
+
+		const connectionUrl = new URL(url);
+		const pathname = connectionUrl.pathname;
+
+		const database = pathname.split('/')[pathname.split('/').length - 1];
+		if (!database) {
+			console.error(
+				'You should specify a database name in connection string (singlestore://USER:PASSWORD@HOST:PORT/DATABASE)',
+			);
+			process.exit(1);
+		}
+		return { database, url };
+	} else {
+		return {
+			database: credentials.database,
+			credentials,
+		};
+	}
+};
+
+export const connectToSingleStore = async (
```

Comment:
By the time of this writing, SingleStore is 100% compatible with the MySQL drive, including the credentials needed and the way they are parsed, so we could return it like
```ts
const parseSingleStoreCredentials = (credentials: SingleStoreCredentials) => const parseMySqlCredentials = (credentials as SingleStoreCredentials)
```
The main reasons that I see in favor of adding a new connection function for SingleStore are type safety since we would need to ensure that `MySQLCredentials` is 100% compatible with `SingleStoreCredentials`,and full autonomy on the way we implement the way we connect to SingleStore, either if `MySql` or `SingleStore` changes the way they connect to DBs

Reply:
Valid